core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/impl/LocalDictDimensionDataChunkStore.java

```diff
@@ -49,6 +51,29 @@ public void putArray(int[] invertedIndex, int[] invertedIndexReverse, byte[] dat
     this.dimensionDataChunkStore.putArray(invertedIndex, invertedIndexReverse, data);
   }
 
+  @Override
+  public void fillVector(int[] invertedIndex, int[] invertedIndexReverse, byte[] data,
+      ColumnVectorInfo vectorInfo) {
+    int columnValueSize = dimensionDataChunkStore.getColumnValueSize();
+    int rowsNum = data.length / columnValueSize;
+    CarbonColumnVector vector = vectorInfo.vector;
+    if (!dictionary.isDictionaryUsed()) {
+      vector.setDictionary(dictionary);
+      dictionary.setDictionaryUsed();
+    }
+    for (int i = 0; i < rowsNum; i++) {
+      int surrogate = CarbonUtil.getSurrogateInternal(data, i * columnValueSize, columnValueSize);
+      if (surrogate == CarbonCommonConstants.MEMBER_DEFAULT_VAL_SURROGATE_KEY) {
+        vector.putNull(i);
+        vector.getDictionaryVector().putNull(i);
+      } else {
+        vector.putNotNull(i);
+        vector.getDictionaryVector().putInt(i, surrogate);
```

Comment:
Can we try to put the surrogate directly and avoid 2 vectors here. For null values we can handle at the time of conversion  from surrogate key to actual value in `CarbonDictionaryWrapper`. If it is feasible can we try this?

Reply:
it is as per old code, will check feasible